modules/logging/formattedencoder.go

```diff
@@ -67,15 +65,28 @@ func (se FormattedEncoder) EncodeEntry(ent zapcore.Entry, fields []zapcore.Field
 	if err != nil {
 		return buf, err
 	}
-	appName := strings.SplitN(ent.LoggerName, ".", 2)[0]
-	// set the vals in the replacer
-	err = jsonparser.ObjectEach(buf.Bytes(), visitor(appName, repl))
-	buf.Reset() // the buffer is only used to collect placeholders' values anyway
-	if err != nil {
-		return buf, err
-	}
+	repl.Map(func(key string) (interface{}, bool) {
+		path := strings.Split(key, ">")
+		value, dataType, _, err := jsonparser.Get(buf.Bytes(), path...)
+		if err != nil {
+			return nil, false
+		}
+		switch dataType {
+		case jsonparser.NotExist:
+			return nil, false
+		case jsonparser.Array, jsonparser.Boolean, jsonparser.Null, jsonparser.Number, jsonparser.Object, jsonparser.String, jsonparser.Unknown:
+			// if a value exists, return it as is. A byte is a byte is a byte. The replacer handles them just fine.
+			return value, true
+		default:
+			return nil, false
+		}
+	})
 
 	out := repl.ReplaceKnown(se.Template, "")
+	// The buffer is only used to find the values of placeholders.
+	// The content has served its purpose. It's time for it to go to repurpose the buffer.
+	buf.Reset()
+
 	// Unescape escaped quotes
 	buf.AppendString(strings.Replace(out, `\"`, `"`, -1))
```

Comment:
Why is this needed?

Reply:
Because strings with quotes inside the JSON end up being printed with the escape backslashes, e.g. `::1 - - [28/May/2020:22:45:40 +0300] \"GET / HTTP/2.0\" 200 78212`, while they only needed to be escaped because they were embedded inside a JSON string.

Reply:
Interesting... so bruger's lib doesn't unescape quotes then?

Reply:
Nope. Honestly, I don't expect it to manipulate the strings. If it did, I'd call it a bug. It should just hand out the content as-is.

Reply:
Make sense. I guess this is fine then! If we find this to be buggy or difficult to maintain, we could always surround the string with quotes and then call `json.Unmarshal()` to do the dirty work for us. But this LGTM for now.